portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,77 @@
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 import java.io.IOException;
+import com.google.sps.data.Comment;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.Arrays;
+import com.google.gson.Gson;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/* Servlet that handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    // Get comments from Datastore
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    // Add comment properties in an ArrayList
+    ArrayList<Comment> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String text = (String) entity.getProperty("text-input");
+      long timestamp = (long) entity.getProperty("timestamp");
+
+      comments.add(new Comment(id, text, timestamp));
+    }
+    // Converts to JSON and responds
+    response.setContentType("application/json;");
+    String json = convertToJsonUsingGson(comments);
+    response.getWriter().println(json);
+  }
+
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form in html page.
+    String text = getComment(request, "text-input");
+    long timestamp = System.currentTimeMillis();
+
+    // Create new Entity for Datastore
+    Entity comEntity = new Entity("Comment");
+    comEntity.setProperty("text-input", text);
+    comEntity.setProperty("timestamp", timestamp);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(comEntity);
+
+    // Redirect back to the HTML page.
+    response.sendRedirect("../about/about.html");
+  }
+
+
+  /* @return the request parameter and trims whitespace */
+  private String getComment(HttpServletRequest request, String name) {
+    String value = request.getParameter(name);
+    return value.trim();
+  }
+
+   /* Method to convert to Json */
+   private String convertToJsonUsingGson(ArrayList<Comment> messages) {
+    Gson gson = new Gson();
+    String json = gson.toJson(messages);
+    return json;
```

Comment:
The code looks good, I have a few notes on the naming:
1) convert *what* to Json?
2) in general, the caller of a function shouldn't need to know that this conversion is being done through Gson, unless you have two functions that do the same thing, and using Gson is important to one of them.
Similarly. your doc string comment should explain what you're converting.